cloudinit/sources/DataSourceAzure.py

```diff
@@ -554,6 +554,35 @@ def _is_ephemeral_networking_up(self) -> bool:
             or self._ephemeral_dhcp_ctx.lease is None
         )
 
+    def _check_azure_proxy_agent_status(self) -> None:
+        """check if azure-proxy-agent is fully running and can procede to
+        communicate with IMDS and Wire Server. If not, fail provisioning.
+        Azure Guest Proxy Agent source code:
+        https://github.com/azure/guestproxyagent"""
+        try:
+            cmd = [
+                "azure-proxy-agent",
+                "--status",
+                "--wait",
+                "120",
+            ]
+            out, err = subp.subp(cmd)
+            report_diagnostic_event(
+                "Running azure-proxy-agent %s resulted"
+                "in stderr output: %s with stdout: %s" % (cmd, err, out),
+                logger_func=LOG.debug,
+            )
+
```

Comment:
nit: remove the extra lines around except blocks for consistency with rest of module